app/controllers/appeal-application/check-and-send.ts

```diff
@@ -137,16 +167,44 @@ function postCheckAndSend(updateAppealService: UpdateAppealService) {
   };
 }
 
-function setupCheckAndSendController(middleware: Middleware[], updateAppealService: UpdateAppealService): Router {
+function getFinishPayment(updateAppealService: UpdateAppealService, paymentService: PaymentService) {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const paymentDetails = JSON.parse(await paymentService.getPaymentDetails(req, req.session.appeal.paymentReference));
+
```

Comment:
@t0nin0s can you introduce some logging of payment status against ccd case id over here? This will be assist debugging logs in Azure once feature goes live.

Reply:
added some logging

Reply:
added